### PR TITLE
Real tests, real life

### PR DESCRIPTION
**Pull request by taras**
Will it run? TBD

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,6 +20,10 @@ commands:
           name: "Restore Yarn cache"
           key: yarn-cache-v3-{{ arch }}-{{ checksum "yarn.lock" }}
 
+      - run:
+          name: "Install appium"
+          command: yarn global add appium
```
Comment (cowboyd) [inline .circleci/config.yml]: Why not just put it in the `devDependencies`?

Comment (taras) [reply .circleci/config.yml]: It has to be installed globally

**Comment by taras** [comment]
Dropping this here for tomorrow https://www.reddit.com/r/androiddev/comments/5mdpd7/circleci_and_android24_emulators/

File: .circleci/config.yml (hunk 1 above)
Comment (cowboyd) [reply .circleci/config.yml]: Yeah, I remember that, but as I recall, I was able to safely ignore that advice. This is actually part of the NativeScript culture I've sensed, that seems to be shared with the general native app development community of not really thinking/prioritizing automation, and distribution (as opposed to massaging a one-off local dev environment into working order and then using it to do everything from development to publishing applications).

In my opinion, every chance we have to push back against those types of decisions that propel an ecosystem towards non-reproducibility of environment, we should... provided it isn't too costly. In this case, I'm pretty sure it works.

https://github.com/cowboyd/nativescript-vacation-tracker/blob/master/package.json#L38

Tests worked just fine with a local environment on that project, but not in CI (but pretty sure it was related to the `hooks/` directory not being persisted.

Comment (taras) [reply .circleci/config.yml]: I removed the global and added local install of appium.

**Comment by taras** [comment]
[Testing with Android emulator on CircleCI 2.0](https://support.circleci.com/hc/en-us/articles/360000028928-Testing-with-Android-emulator-on-CircleCI-2-0)

> Running the Android emulator is not currently supported on CircleCI 2.0, since it's not supported by the type of virtualization CircleCI uses on Linux. We recommend using an external service such as Firebase Test Lab to run your emulator tests from your CircleCI job.

- [Flank: Smart Test Runner for Firebase](https://link.medium.com/BxPEEW4PZU)
- [Android Emulators on Travis-CI](https://link.medium.com/8vg5rqwRZU)

--

- [Android in docker solution with noVNC supported and video recording](https://github.com/budtmo/docker-android)
- [Running Docker Android on Azure, AWS or GCE](https://github.com/budtmo/docker-android/blob/master/README_CLOUD.md)

**Comment by cowboyd** [comment]
Closed. It's cheaper by far to use a simulator service than to pay senior engineering staff to stand up simulators inside Azure containers.

**Pull request closed without merging**